geomstats/information_geometry/multivariate_normal.py

```diff
@@ -0,0 +1,348 @@
+"""Statistical Manifold of multivaraite normal distributions with the Fisher metric.
+
+Lead author: Antoine Collas.
+"""
+
+import math
+
+from scipy.stats import multivariate_normal, norm
+
+import geomstats
+import geomstats.backend as gs
+from geomstats.geometry.base import OpenSet
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+from geomstats.information_geometry.base import InformationManifoldMixin
+from geomstats.information_geometry.normal import NormalMetric
+
+
+class MultivariateDiagonalNormalDistributions(OpenSet, InformationManifoldMixin):
+    """Class for the manifold of diagonal multivariate normal distributions."""
+
+    def __init__(self, n):
+        self.n = n
+        self.euclidean_n = Euclidean(dim=n)
+        dim = int(2 * n)
+        super().__init__(dim=dim, embedding_space=Euclidean(dim))
+        self.metric = MultivariateDiagonalNormalMetric(n=n)
+        self._unstack_location_diagonal = self.metric._unstack_location_diagonal
+        self._stack_location_diagonal = self.metric._stack_location_diagonal
+
+    def belongs(self, point, atol=gs.atol):
+        """Evaluate if the point belongs to the manifold.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 2*n]
+            Point to test.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the space.
+        """
+        point_dim = point.shape[-1]
+        belongs = point_dim == self.dim
+        _, diagonal = self._unstack_location_diagonal(point)
+        belongs = gs.logical_and(belongs, gs.all(diagonal >= atol, axis=-1))
+        return belongs
+
+    def random_point(self, n_samples=1):
+        """Sample parameters of normal distributions.
```

Comment:
Maybe "Generate random parameters of multivariate diagonal normal distributions" would be less confusing ?